scrapy/http/request/__init__.py

```diff
@@ -187,7 +187,7 @@ def to_dict(self, *, spider: Optional["scrapy.Spider"] = None) -> dict:
         }
         for attr in self.attributes:
             d.setdefault(attr, getattr(self, attr))
-        if type(self) is not Request:
+        if not isinstance(self, Request):
```

Comment:
Is this the one breaking the tests? to_dict fails on the pipeline and also isinstance caters to inheritance, whilst type does not. [https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance](Difference type and isinstance)

```suggestion
        if type(self) is not Request:
```